### DISCUSSION TITLE
Support touch ID on macos

### DISCUSSION THREAD

**Discussion by tony-sol** [wezterm/wezterm#4485] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

Native macos Terminal.app can support Touch ID on sudo operations via 
```
 auth       sufficient     pam_tid.so
```
in `/etc/pam.d/sudo`, but it seems that not working in wezterm

Are there plans to make it possible to support touch-id/windows hello/etc. in future releases?

**Comment by nekowinston** [comment]
This works for me (wezterm 20230712-072601-f4abf8fd):
![2023-10-24-223703@2x](https://github.com/wez/wezterm/assets/79978224/52dc72df-aa96-4a20-a24f-8eabef32278b)

**Reply by tony-sol** [reply]
Wow
Is there any extra config required? Can you share your lua config?
I can't make this work only with pam.d/sudo changes :c

**Reply by wez** [reply]
wezterm doesn't participate in PAM; nothing in wezterm's code or configuration should impact this

**Reply by tony-sol** [reply, marked as answer]
After a few tests, found that `tmux` as `default_prog` breaks this feature.
By default, tmux operates its own sessions and can't properly reattach to user's login session.

Installing this https://github.com/fabianishere/pam_reattach, along with modify `/etc/pom.d/sudo` as

```
auth       optional       /opt/homebrew/lib/pam/pam_reattach.so
auth       sufficient     pam_tid.so
```
works both in terminal.app and wezterm, both inside and outside tmux, like a charm.

Thanks!

**Reply by muHashh** [reply]
Old thread but what font do you use @nekowinston?

**Reply by abhijeetviswa** [reply]
For folks who are confused why this doesn't work, the path is `/etc/pam.d/sudo` i.e `pam.d` not `pom.d`.
